core/src/com/unciv/models/ruleset/Unique.kt

```diff
@@ -14,16 +14,76 @@ import com.unciv.models.translations.hasPlaceholderParameters
 import com.unciv.ui.worldscreen.unit.UnitActions
 import kotlin.random.Random
 
-class Unique(val text:String){
+
+// parameterName values should be compliant with autogenerated values in TranslationFileWriter.generateStringsFromJSONs
+// Eventually we'll merge the translation generation to take this as the source of that
+enum class UniqueParameterType(val parameterName:String, val complianceCheck:(String, Ruleset)->Boolean) {
+    Number("amount", { s, r -> s.toIntOrNull() != null }),
+    UnitFilter("unitType", { s, r -> r.unitTypes.containsKey(s) || unitTypeStrings.contains(s) }),
+    Unknown("",{s,r -> true});
+
+    companion object {
+        val unitTypeStrings = hashSetOf(
+            "Military",
+            "Civilian",
+            "non-air",
+            "relevant",
+            "Nuclear Weapon",
+            "City",
+            // These are up for debate
+            "Air",
+            "land units",
+            "water units",
+            "air units",
+            "military units",
+            "submarine units",
+            // Note: this can't handle combinations of parameters (e.g. [{Military} {Water}])
+        )
+    }
+}
+
+enum class UniqueType(val text:String){
+
+    ConsumesResources("Consumes [amount] [resource]");
+
+    /** For uniques that have "special" parameters that can accept multiple types, we can override them manually
+     *  For 95% of cases, auto-matching is fine. */
+    val parameterTypeMap= ArrayList<List<UniqueParameterType>>()
+    init {
+        for (placeholder in text.getPlaceholderParameters()) {
+            val matchingParameterType =
+                UniqueParameterType.values().firstOrNull { it.parameterName == placeholder }
+            if (matchingParameterType != null) parameterTypeMap.add(listOf(matchingParameterType))
```

Comment:
Prime Elvis the King example: `parameterTypeMap.add(listOf(matchingParameterType ?: UniqueParameterType.Unknown))` - even a oneliner dropping matchingParameterType could be readable... Just teasing.